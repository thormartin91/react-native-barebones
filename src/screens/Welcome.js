/** @flow */

'use strict'

import React from 'react';
import { StyleSheet, View } from 'react-native';
import { Actions } from 'react-native-router-flux';
import { Colors } from '../components/Colors';
import Header from '../components/Header';
import Text from '../components/Text';
import Button from '../components/Button';

class Welcome extends React.Component {

  render() {
    return (
      <View style={styles.container}>
        <Header>
          Hello, world!
        </Header>
        <Text style={styles.instructions}>
          To get started, edit {'\n'} app/screens/Welcome.js {'\n'}
          {'\n'}
          Press Cmd+R to reload, {'\n'} Cmd+D or shake for dev menu
        </Text>
        <Button
          onPress={() => Actions.counter({
            onBack: () => Actions.pop() })}
            caption={"Counter"}
        ></Button>
      </View>
    );
  }

}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: Colors.lightBlue,
  },
  instructions: {
    textAlign: 'center',
    color: Colors.darkGrey,
    marginBottom: 10,
  }
});

export default Welcome;
