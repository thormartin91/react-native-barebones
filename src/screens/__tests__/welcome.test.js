'use strict'

import 'react-native';
import React from 'react';
import Welcome from '../Welcome';
import renderer from 'react-test-renderer';


describe('Welcome-screen', () => {

  it('renders correctly', () => {
    const tree = renderer.create(
      <Welcome />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
