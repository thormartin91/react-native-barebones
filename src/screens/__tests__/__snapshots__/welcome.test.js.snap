exports[`Welcome-screen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "rgba(32,32,32,1)",
          "fontSize": 25,
          "margin": 10,
          "textAlign": "center",
        },
        undefined,
      ]
    }>
    Hello, world!
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "rgba(32,32,32,1)",
        },
        Object {
          "color": "rgba(32,32,32,1)",
          "marginBottom": 10,
          "textAlign": "center",
        },
      ]
    }>
    To get started, edit 
    
    
     app/screens/Welcome.js 
    
    
    
    
    Press Cmd+R to reload, 
    
    
     Cmd+D or shake for dev menu
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "lightgray",
        "height": 30,
        "justifyContent": "center",
        "margin": 3,
        "opacity": 1,
        "padding": 10,
        "width": 100,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgba(32,32,32,1)",
          },
          Object {
            "color": "rgba(32,32,32,1)",
          },
        ]
      }>
      Counter
    </Text>
  </View>
</View>
`;
